src/map/map.cpp

```diff
@@ -3089,8 +3089,8 @@ int map_random_dir(struct block_list *bl, int16 *x, int16 *y)
 	if (dist < 1) dist =1;
 
 	do {
-		short j = 1 + 2*(rnd()%4); //Pick a random diagonal direction
-		short segment = 1+(rnd()%dist); //Pick a random interval from the whole vector in that direction
+		short j = 1 + 2*(rnd(4)); //Pick a random diagonal direction
```

Comment:
Should use DIR_* constant.
Also I would prefer to convert that into a static array/vector and pick a random entry from there.